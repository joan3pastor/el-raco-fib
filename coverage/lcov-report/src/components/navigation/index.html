<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/components/navigation</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="..\..\..\prettify.css" />
    <link rel="stylesheet" href="..\..\..\base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(..\..\..\sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="..\..\..\index.html">All files</a> src/components/navigation
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">78.01% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>337/432</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">86.43% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>191/221</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">54.61% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>77/141</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">75.74% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>281/371</span>
      </div>
    </div>
    <p class="quiet">
      Press <em>n</em> or <em>j</em> to go to the next uncovered block, <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
    </p>
  </div>
  <div class='status-line medium'></div>
<div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file high" data-value="Assignatures.js"><a href="Assignatures.js.html">Assignatures.js</a></td>
	<td data-value="82.35" class="pic high"><div class="chart"><div class="cover-fill" style="width: 82%;"></div><div class="cover-empty" style="width:18%;"></div></div></td>
	<td data-value="82.35" class="pct high">82.35%</td>
	<td data-value="34" class="abs high">28/34</td>
	<td data-value="90.91" class="pct high">90.91%</td>
	<td data-value="22" class="abs high">20/22</td>
	<td data-value="75" class="pct medium">75%</td>
	<td data-value="16" class="abs medium">12/16</td>
	<td data-value="82.35" class="pct high">82.35%</td>
	<td data-value="34" class="abs high">28/34</td>
	</tr>

<tr>
	<td class="file high" data-value="Avisos.js"><a href="Avisos.js.html">Avisos.js</a></td>
	<td data-value="81.98" class="pic high"><div class="chart"><div class="cover-fill" style="width: 81%;"></div><div class="cover-empty" style="width:19%;"></div></div></td>
	<td data-value="81.98" class="pct high">81.98%</td>
	<td data-value="172" class="abs high">141/172</td>
	<td data-value="91.45" class="pct high">91.45%</td>
	<td data-value="117" class="abs high">107/117</td>
	<td data-value="54.29" class="pct medium">54.29%</td>
	<td data-value="35" class="abs medium">19/35</td>
	<td data-value="77.17" class="pct medium">77.17%</td>
	<td data-value="127" class="abs medium">98/127</td>
	</tr>

<tr>
	<td class="file low" data-value="Drawer.js"><a href="Drawer.js.html">Drawer.js</a></td>
	<td data-value="47.37" class="pic low"><div class="chart"><div class="cover-fill" style="width: 47%;"></div><div class="cover-empty" style="width:53%;"></div></div></td>
	<td data-value="47.37" class="pct low">47.37%</td>
	<td data-value="19" class="abs low">9/19</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="8" class="abs high">8/8</td>
	<td data-value="30.77" class="pct low">30.77%</td>
	<td data-value="13" class="abs low">4/13</td>
	<td data-value="47.37" class="pct low">47.37%</td>
	<td data-value="19" class="abs low">9/19</td>
	</tr>

<tr>
	<td class="file high" data-value="Events.js"><a href="Events.js.html">Events.js</a></td>
	<td data-value="81.03" class="pic high"><div class="chart"><div class="cover-fill" style="width: 81%;"></div><div class="cover-empty" style="width:19%;"></div></div></td>
	<td data-value="81.03" class="pct high">81.03%</td>
	<td data-value="58" class="abs high">47/58</td>
	<td data-value="67.65" class="pct medium">67.65%</td>
	<td data-value="34" class="abs medium">23/34</td>
	<td data-value="86.67" class="pct high">86.67%</td>
	<td data-value="15" class="abs high">13/15</td>
	<td data-value="78.43" class="pct medium">78.43%</td>
	<td data-value="51" class="abs medium">40/51</td>
	</tr>

<tr>
	<td class="file low" data-value="ExternalLinks.js"><a href="ExternalLinks.js.html">ExternalLinks.js</a></td>
	<td data-value="20" class="pic low"><div class="chart"><div class="cover-fill" style="width: 20%;"></div><div class="cover-empty" style="width:80%;"></div></div></td>
	<td data-value="20" class="pct low">20%</td>
	<td data-value="15" class="abs low">3/15</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="14.29" class="pct low">14.29%</td>
	<td data-value="14" class="abs low">2/14</td>
	<td data-value="20" class="pct low">20%</td>
	<td data-value="15" class="abs low">3/15</td>
	</tr>

<tr>
	<td class="file low" data-value="Feedback.js"><a href="Feedback.js.html">Feedback.js</a></td>
	<td data-value="40" class="pic low"><div class="chart"><div class="cover-fill" style="width: 40%;"></div><div class="cover-empty" style="width:60%;"></div></div></td>
	<td data-value="40" class="pct low">40%</td>
	<td data-value="5" class="abs low">2/5</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="25" class="pct low">25%</td>
	<td data-value="4" class="abs low">1/4</td>
	<td data-value="40" class="pct low">40%</td>
	<td data-value="5" class="abs low">2/5</td>
	</tr>

<tr>
	<td class="file high" data-value="Horari.js"><a href="Horari.js.html">Horari.js</a></td>
	<td data-value="89.86" class="pic high"><div class="chart"><div class="cover-fill" style="width: 89%;"></div><div class="cover-empty" style="width:11%;"></div></div></td>
	<td data-value="89.86" class="pct high">89.86%</td>
	<td data-value="69" class="abs high">62/69</td>
	<td data-value="86.11" class="pct high">86.11%</td>
	<td data-value="36" class="abs high">31/36</td>
	<td data-value="69.23" class="pct medium">69.23%</td>
	<td data-value="13" class="abs medium">9/13</td>
	<td data-value="90.91" class="pct high">90.91%</td>
	<td data-value="66" class="abs high">60/66</td>
	</tr>

<tr>
	<td class="file high" data-value="Labs.js"><a href="Labs.js.html">Labs.js</a></td>
	<td data-value="81.25" class="pic high"><div class="chart"><div class="cover-fill" style="width: 81%;"></div><div class="cover-empty" style="width:19%;"></div></div></td>
	<td data-value="81.25" class="pct high">81.25%</td>
	<td data-value="16" class="abs high">13/16</td>
	<td data-value="50" class="pct medium">50%</td>
	<td data-value="2" class="abs medium">1/2</td>
	<td data-value="66.67" class="pct medium">66.67%</td>
	<td data-value="9" class="abs medium">6/9</td>
	<td data-value="85.71" class="pct high">85.71%</td>
	<td data-value="14" class="abs high">12/14</td>
	</tr>

<tr>
	<td class="file medium" data-value="Noticies.js"><a href="Noticies.js.html">Noticies.js</a></td>
	<td data-value="71.43" class="pic medium"><div class="chart"><div class="cover-fill" style="width: 71%;"></div><div class="cover-empty" style="width:29%;"></div></div></td>
	<td data-value="71.43" class="pct medium">71.43%</td>
	<td data-value="42" class="abs medium">30/42</td>
	<td data-value="50" class="pct medium">50%</td>
	<td data-value="2" class="abs medium">1/2</td>
	<td data-value="45" class="pct low">45%</td>
	<td data-value="20" class="abs low">9/20</td>
	<td data-value="71.05" class="pct medium">71.05%</td>
	<td data-value="38" class="abs medium">27/38</td>
	</tr>

<tr>
	<td class="file high" data-value="SalesBibl.js"><a href="SalesBibl.js.html">SalesBibl.js</a></td>
	<td data-value="100" class="pic high"><div class="chart"><div class="cover-fill cover-full" style="width: 100%;"></div><div class="cover-empty" style="width:0%;"></div></div></td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="2" class="abs high">2/2</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="2" class="abs high">2/2</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="2" class="abs high">2/2</td>
	</tr>

</tbody>
</table>
</div><div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Fri Jul 17 2020 06:12:00 GMT+0200 (GMT+02:00)
</div>
</div>
<script src="..\..\..\prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="..\..\..\sorter.js"></script>
<script src="..\..\..\block-navigation.js"></script>
</body>
</html>
