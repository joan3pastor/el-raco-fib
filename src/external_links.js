// OAUTH:
export const authURL = 'https://api.fib.upc.edu/v2/o/authorize';
export const tokenURL = 'https://api.fib.upc.edu/v2/o/token';
export const clientID = '8IVfHurpzfHhC68LdfpIoRoh7uWcr5bXBkCGroV4';
export const clientSECRET = '8W3Xbf3kRfIQK3rW8QR2xED3TxZ1cI3kMv4VnqyeoreruGNKXBnL8MtcJUztndI7QiuqvJhkvfNE53fOSfbqOT4DEH75d2TmgrmPTHeU5HDuRrA2tFlyYSBoKH31O9K4';
export const callbackURI = 'apifib://callback'
export const revokeURL = "https://api.fib.upc.edu/v2/o/revoke_token/";

// API URLs:
export const joURL = 'https://api.fib.upc.edu/v2/jo/';
export const fotoURL = 'https://api.fib.upc.edu/v2/jo/foto.jpg';
export const avisosURL = 'https://api.fib.upc.edu/v2/jo/avisos/';
export const horariURL = 'https://api.fib.upc.edu/v2/jo/classes/';
export const assigURL = 'https://api.fib.upc.edu/v2/jo/assignatures/';
export const noticiesURL = 'https://api.fib.upc.edu/v2/noticies/';
export const imglabsURL = 'https://api.fib.upc.edu/v2/laboratoris/imatges/';
export const eventsURL = 'https://api.fib.upc.edu/v2/events/';
export const practiquesURL = 'https://api.fib.upc.edu/v2/jo/practiques/';
export const placesURL = 'https://api.fib.upc.edu/v2/assignatures/places/';
export const mobilitatURL = 'https://api.fib.upc.edu/v2/universitats-erasmus/';
export const lecturesURL = 'https://api.fib.upc.edu/v2/lectures/properes/';
export const QActualURL = 'https://api.fib.upc.edu/v2/quadrimestres/actual/';
export const aulesURL = 'https://api.fib.upc.edu/v2/aules/';

// UPC Notícies RSS Channel
export const upcNoticiesRSSca = "https://www.upc.edu/ca/sala-de-premsa/noticies/RSS";
export const upcNoticiesRSSes = "https://www.upc.edu/ca/sala-de-premsa/noticies/RSS";
export const upcNoticiesRSSen = "https://www.upc.edu/ca/sala-de-premsa/noticies/RSS";

// Reservas Bibl. 
export const salesBiblioteca = 'https://apps.bibliotecnica.upc.edu/reserva_sales/comunes/disponibilidad.php?Id_centro=1&Admin=1&Parametros=TnVtX2NvbHVtbmFzPTYmQWx0bz0mQW5jaG89MTAwJSZGdWVudGU9QXJpYWwmVGFtYW5vX2Z1ZW50ZT0xOHB4JlRhbWFub19jb250b3Jubz0ycHgmTWFyZ2VuX2RvY3VtZW50bz0wcHgmQ29sb3JfZm9uZG89ZmZmZmZmJlBhZGRpbmdfY2VsZGE9MTBweCZNb3N0cmFyX3RpdHVsbz0xJkFicmV2aWF0dXJhcz0xJk1vc3RyYXJfbGlua19hZG1pbj0wJk1vc3RyYXJfdGV4dG9fZGlzcG9uaWJsZXM9MCZDb21wYWN0YXJfdWJpY2FjaW9uZXM9MCZNb3N0cmFyX2VubGFjZV9pbmZvcm1hY2lvbj0wJk1vc3RyYXJfYm90b25fcmVzZXJ2YXI9MQ==';

// Enllaços d'interès
export const linkDEFIB = 'http://defib.upc.edu/qui_som';
export const linkAssociacions = 'https://www.fib.upc.edu/ca/la-fib/vida-universitaria/associacions';
export const linkEduroam = 'https://serveistic.upc.edu/ca/wifiupc/documentacio/assistent-de-configuracio-per-eduroam';
export const linkCorreu = 'https://www.fib.upc.edu/ca/la-fib/serveis-tic/correu-destudiant';
export const linkImpressio = 'https://www.fib.upc.edu/ca/la-fib/serveis-tic/servei-dimpressio';
export const linkUPClink = 'https://www.fib.upc.edu/ca/la-fib/serveis-tic/vpn-upclink';
export const linkAcordsUPC = 'https://serveistic.upc.edu/ca/distsoft/el-servei/programari-per-a-estudiantsi-professors';
export const linkVMware = 'https://raco.fib.upc.edu/vmware';
export const linkGithub = 'https://education.github.com/pack';
export const linkAzure = 'https://azure.microsoft.com/en-us/free/students/';
export const linkOffice = 'https://www.microsoft.com/es-es/education/products/office';
export const linkGCP = 'https://cloud.google.com/edu/';

// Feedback
export const playstorelink = 'https://play.google.com/store/apps/details?id=com.fibupc';

// Programes de Mobilitat
export const mobInfoGeneral = 'https://www.fib.upc.edu/ca/mobilitat/outgoing/estades-academiques';
export const mobSessionsInfo = 'https://www.fib.upc.edu/ca/mobilitat/outgoing/sessions-informatives';
export const mobCalendari = 'https://www.fib.upc.edu/ca/mobilitat/outgoing/calendari-mobilitat';
export const mobBeques = 'https://www.upc.edu/sga/ca/Beques/BequesMobilitat';

// Github
export const githubURL = "https://github.com/joan3pastor/el-raco-fib";
