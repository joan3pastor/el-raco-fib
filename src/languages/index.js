import {ca} from "./ca.js";
import {es} from "./es.js";
import {en} from "./en.js";

export default {
    ca: ca,
    es: es,
    en: en,
};