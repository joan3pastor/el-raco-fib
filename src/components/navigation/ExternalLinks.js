import React, { Component } from 'react';
import { Text, View, ScrollView, TouchableOpacity, Linking } from 'react-native';
import { connect } from 'react-redux';

import {  } from '../../actions';
import {linkDEFIB, linkAssociacions, linkEduroam, linkCorreu, linkImpressio, linkUPClink, linkAcordsUPC, linkVMware, linkGithub, linkAzure, linkOffice, linkGCP} from "../../external_links";

const styleLinks = {color:'#0645AD', fontWeight:'bold', paddingTop:10, paddingBottom:5, paddingRight:10};

export class ExternalLinks extends Component {

  render() {

    return (
      <ScrollView style={{flex:1, backgroundColor:'#fff', overflow:'hidden'}}>
            <View style={{padding:15, backgroundColor:'#f2f2f2', elevation:1.5}}>
            <Text style={{color:'#202020', fontSize:20, fontWeight:'400'}}>{`${this.props.lang.title1}   `}</Text>
            </View>
            <View style={{marginLeft:20, backgroundColor: '#fff'}}>

                <TouchableOpacity onPress={() => Linking.openURL(linkDEFIB)}>
                    <Text style={{color:'#0645AD', fontWeight:'bold', paddingTop:10, paddingBottom:0}}>{`${this.props.lang.A1E1}   `}</Text>
                </TouchableOpacity>
                <Text>{`${this.props.lang.A1E1Comentari}   `}</Text>

                <TouchableOpacity onPress={() => Linking.openURL(linkAssociacions)}>
                    <Text style={styleLinks}>{`${this.props.lang.A1E2}   `}</Text>
                </TouchableOpacity>

            </View>
            <View style={{padding:15, backgroundColor:'#f2f2f2', marginTop:25, elevation:1.5}}>
                <Text style={{color:'#202020', fontSize:20, fontWeight:'400'}}>{`${this.props.lang.title2}   `}</Text>
            </View>
            <View style={{marginLeft:20, backgroundColor: '#fff'}}>

                <TouchableOpacity onPress={() => Linking.openURL(linkEduroam)}>
                    <Text style={styleLinks}>{`${this.props.lang.A2E1}   `}</Text>
                </TouchableOpacity>
                <TouchableOpacity onPress={() => Linking.openURL(linkCorreu)}>
                    <Text style={styleLinks}>{`${this.props.lang.A2E2}   `}</Text>
                </TouchableOpacity>
                <TouchableOpacity onPress={() => Linking.openURL(linkImpressio)}>
                    <Text style={styleLinks}>{`${this.props.lang.A2E3}   `}</Text>
                </TouchableOpacity>
                <TouchableOpacity onPress={() => Linking.openURL(linkUPClink)}>
                    <Text style={styleLinks}>{`${this.props.lang.A2E4}   `}</Text>
                </TouchableOpacity>

            </View>
            <View style={{padding:15, backgroundColor:'#f2f2f2', marginTop:25, elevation:1.5}}>
                <Text style={{color:'#202020', fontSize:20, fontWeight:'400'}}>{`${this.props.lang.title3}   `}</Text>
            </View>
            <View style={{marginLeft:20, backgroundColor: '#fff'}}>

                <TouchableOpacity onPress={() => Linking.openURL(linkAcordsUPC)}>
                    <Text style={styleLinks}>{`${this.props.lang.A3E1}   `}</Text>
                </TouchableOpacity>
                <TouchableOpacity onPress={() => Linking.openURL(linkVMware)}>
                    <Text style={styleLinks}>{`${this.props.lang.A3E2}   `}</Text>
                </TouchableOpacity>
                <TouchableOpacity onPress={() => Linking.openURL(linkGithub)}>
                    <Text style={styleLinks}>{`${this.props.lang.A3E3}   `}</Text>
                </TouchableOpacity>
                <TouchableOpacity onPress={() => Linking.openURL(linkAzure)}>
                    <Text style={styleLinks}>{`${this.props.lang.A3E4}   `}</Text>
                </TouchableOpacity>
                <TouchableOpacity onPress={() => Linking.openURL(linkOffice)}>
                    <Text style={styleLinks}>{`${this.props.lang.A3E5}   `}</Text>
                </TouchableOpacity>
                <TouchableOpacity onPress={() => Linking.openURL(linkGCP)}>
                    <Text style={styleLinks}>{`${this.props.lang.A3E6}   `}</Text>
                </TouchableOpacity>

            </View>
            <Text style={{padding:12, marginTop:20}}>{`${this.props.lang.msgManteniment}   `}</Text>
      </ScrollView>
    );

  }

}

function mapStateToProps(state) {
  return {
      lang: state.lang.links,
  };
}

export default connect(mapStateToProps, {  })(ExternalLinks);
