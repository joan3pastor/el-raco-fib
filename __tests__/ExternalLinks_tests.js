import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import React from 'react';
import configureStore from 'redux-mock-store';
import { shallow } from 'enzyme';
import axios from 'axios'

// LANG
import {ca} from "../src/languages/ca";
const lang = ca.links;

import {ExternalLinks} from "../src/components/navigation/ExternalLinks";
import {linkDEFIB, linkAssociacions, linkEduroam, linkCorreu, linkImpressio, linkUPClink, linkAcordsUPC, linkVMware, linkGithub, linkAzure, linkOffice, linkGCP} from "../src/external_links";

Enzyme.configure({ adapter: new Adapter() });
const middlewares = []; // you can mock any middlewares here if necessary
const mockStore = configureStore(middlewares);

const initialState = {};

beforeEach( () => {})

describe('Testing ExternalLinks component', () => {
  it('renders as expected', () => {
    const wrapper = shallow(
      <ExternalLinks store={mockStore(initialState)} lang={lang} />
    );
    expect(wrapper.dive().dive()).toMatchSnapshot();
  });
  // it("linkDEFIB url returns 200[OK] code", async () => {
  //   resp = await axios.get(linkDEFIB);
  //   expect(resp.status == 200).toBe(true);
  // });
  it("linkAssociacions url returns 200[OK] code", async () => {
    resp = await axios.get(linkAssociacions);
    expect(resp.status == 200).toBe(true);
  });
  it("linkEduroam url returns 200[OK] code", async () => {
    resp = await axios.get(linkEduroam);
    expect(resp.status == 200).toBe(true);
  });
  it("linkCorreu url returns 200[OK] code", async () => {
    resp = await axios.get(linkCorreu);
    expect(resp.status == 200).toBe(true);
  });
  it("linkImpressio url returns 200[OK] code", async () => {
    resp = await axios.get(linkImpressio);
    expect(resp.status == 200).toBe(true);
  });
  it("linkUPClink url returns 200[OK] code", async () => {
    resp = await axios.get(linkUPClink);
    expect(resp.status == 200).toBe(true);
  });
  it("linkAcordsUPC url returns 200[OK] code", async () => {
    resp = await axios.get(linkAcordsUPC);
    expect(resp.status == 200).toBe(true);
  });
  it("linkVMware url returns 200[OK] code", async () => {
    resp = await axios.get(linkVMware);
    expect(resp.status == 200).toBe(true);
  });
  it("linkGithub url returns 200[OK] code", async () => {
    resp = await axios.get(linkGithub);
    expect(resp.status == 200).toBe(true);
  });
  it("linkAzure url returns 200[OK] code", async () => {
    resp = await axios.get(linkAzure);
    expect(resp.status == 200).toBe(true);
  });
  it("linkOffice url returns 200[OK] code", async () => {
    resp = await axios.get(linkOffice);
    expect(resp.status == 200).toBe(true);
  });
  it("linkGCP url returns 200[OK] code", async () => {
    resp = await axios.get(linkGCP);
    expect(resp.status == 200).toBe(true);
  });
});

