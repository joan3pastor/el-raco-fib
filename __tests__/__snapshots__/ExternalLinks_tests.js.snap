// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing ExternalLinks component renders as expected 1`] = `
<RCTScrollView
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 1,
      "overflow": "hidden",
    }
  }
>
  <View>
    <View
      style={
        Object {
          "backgroundColor": "#f2f2f2",
          "elevation": 1.5,
          "padding": 15,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#202020",
            "fontSize": 20,
            "fontWeight": "400",
          }
        }
      >
        La Facultat   
      </Text>
    </View>
    <View
      style={
        Object {
          "backgroundColor": "#fff",
          "marginLeft": 20,
        }
      }
    >
      <TouchableOpacity
        activeOpacity={0.2}
        onPress={[Function]}
      >
        <Text
          style={
            Object {
              "color": "#0645AD",
              "fontWeight": "bold",
              "paddingBottom": 0,
              "paddingTop": 10,
            }
          }
        >
          DEFIB (Delegació d'Estudiants de la FIB)   
        </Text>
      </TouchableOpacity>
      <Text>
           - Bona secció de FAQ per a nous estudiants.   
      </Text>
      <TouchableOpacity
        activeOpacity={0.2}
        onPress={[Function]}
      >
        <Text
          style={
            Object {
              "color": "#0645AD",
              "fontWeight": "bold",
              "paddingBottom": 5,
              "paddingRight": 10,
              "paddingTop": 10,
            }
          }
        >
          Associacions   
        </Text>
      </TouchableOpacity>
    </View>
    <View
      style={
        Object {
          "backgroundColor": "#f2f2f2",
          "elevation": 1.5,
          "marginTop": 25,
          "padding": 15,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#202020",
            "fontSize": 20,
            "fontWeight": "400",
          }
        }
      >
        Guies   
      </Text>
    </View>
    <View
      style={
        Object {
          "backgroundColor": "#fff",
          "marginLeft": 20,
        }
      }
    >
      <TouchableOpacity
        activeOpacity={0.2}
        onPress={[Function]}
      >
        <Text
          style={
            Object {
              "color": "#0645AD",
              "fontWeight": "bold",
              "paddingBottom": 5,
              "paddingRight": 10,
              "paddingTop": 10,
            }
          }
        >
          Manual de connexió a Eduroam (Wi-Fi)   
        </Text>
      </TouchableOpacity>
      <TouchableOpacity
        activeOpacity={0.2}
        onPress={[Function]}
      >
        <Text
          style={
            Object {
              "color": "#0645AD",
              "fontWeight": "bold",
              "paddingBottom": 5,
              "paddingRight": 10,
              "paddingTop": 10,
            }
          }
        >
          Configuració del correu d'estudiant   
        </Text>
      </TouchableOpacity>
      <TouchableOpacity
        activeOpacity={0.2}
        onPress={[Function]}
      >
        <Text
          style={
            Object {
              "color": "#0645AD",
              "fontWeight": "bold",
              "paddingBottom": 5,
              "paddingRight": 10,
              "paddingTop": 10,
            }
          }
        >
          Servei d'impressió   
        </Text>
      </TouchableOpacity>
      <TouchableOpacity
        activeOpacity={0.2}
        onPress={[Function]}
      >
        <Text
          style={
            Object {
              "color": "#0645AD",
              "fontWeight": "bold",
              "paddingBottom": 5,
              "paddingRight": 10,
              "paddingTop": 10,
            }
          }
        >
          Configuració UPClink (VPN per accedir als servidors de la FIB des de fora la facultat)   
        </Text>
      </TouchableOpacity>
    </View>
    <View
      style={
        Object {
          "backgroundColor": "#f2f2f2",
          "elevation": 1.5,
          "marginTop": 25,
          "padding": 15,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#202020",
            "fontSize": 20,
            "fontWeight": "400",
          }
        }
      >
        Programari i Serveis per a l'Estudiant   
      </Text>
    </View>
    <View
      style={
        Object {
          "backgroundColor": "#fff",
          "marginLeft": 20,
        }
      }
    >
      <TouchableOpacity
        activeOpacity={0.2}
        onPress={[Function]}
      >
        <Text
          style={
            Object {
              "color": "#0645AD",
              "fontWeight": "bold",
              "paddingBottom": 5,
              "paddingRight": 10,
              "paddingTop": 10,
            }
          }
        >
          Acords de distribució de programari UPC (Microsoft Imagine, Autocad, Matlab...)   
        </Text>
      </TouchableOpacity>
      <TouchableOpacity
        activeOpacity={0.2}
        onPress={[Function]}
      >
        <Text
          style={
            Object {
              "color": "#0645AD",
              "fontWeight": "bold",
              "paddingBottom": 5,
              "paddingRight": 10,
              "paddingTop": 10,
            }
          }
        >
          VMware Academic Program (Ofert per la FIB)   
        </Text>
      </TouchableOpacity>
      <TouchableOpacity
        activeOpacity={0.2}
        onPress={[Function]}
      >
        <Text
          style={
            Object {
              "color": "#0645AD",
              "fontWeight": "bold",
              "paddingBottom": 5,
              "paddingRight": 10,
              "paddingTop": 10,
            }
          }
        >
          Github Student Developer Pack (Github private projects, ASW Educate, Heroku Free Dynos, etc.)   
        </Text>
      </TouchableOpacity>
      <TouchableOpacity
        activeOpacity={0.2}
        onPress={[Function]}
      >
        <Text
          style={
            Object {
              "color": "#0645AD",
              "fontWeight": "bold",
              "paddingBottom": 5,
              "paddingRight": 10,
              "paddingTop": 10,
            }
          }
        >
          Microsoft Azure for Students   
        </Text>
      </TouchableOpacity>
      <TouchableOpacity
        activeOpacity={0.2}
        onPress={[Function]}
      >
        <Text
          style={
            Object {
              "color": "#0645AD",
              "fontWeight": "bold",
              "paddingBottom": 5,
              "paddingRight": 10,
              "paddingTop": 10,
            }
          }
        >
          Microsoft Office for Students + 1 TB OneDrive   
        </Text>
      </TouchableOpacity>
      <TouchableOpacity
        activeOpacity={0.2}
        onPress={[Function]}
      >
        <Text
          style={
            Object {
              "color": "#0645AD",
              "fontWeight": "bold",
              "paddingBottom": 5,
              "paddingRight": 10,
              "paddingTop": 10,
            }
          }
        >
          Google Cloud Platform Education Credits   
        </Text>
      </TouchableOpacity>
    </View>
    <Text
      style={
        Object {
          "marginTop": 20,
          "padding": 12,
        }
      }
    >
      Si algun enllaç deixa de funcionar envieu-me un correu des de l'apartat "Feedback & Bugs".   
    </Text>
  </View>
</RCTScrollView>
`;
